src/traverse.js

```diff
@@ -89,6 +96,7 @@ export function traverse(schema, options, spec, context) {
     }
   }
 
+  seenSchemasStack.pop();
```

Comment:
I've noticed you push to the stack conditionally (only if there is a context) but pop unconditionally. There must be something wrong with this.

Reply:
you were right, good 👁️ , sorry for that